### PR TITLE
Auth Refactor 

### PR DESCRIPTION
**Pull request by neymarjimoh**
**What does this PR do?**
 - Auth controller and route refactor
- addition of product and investment collection
- update documentation based on changes
- add new investor fields to user schema
- added validation rules for required endpoints 

**description of Task to be completed?**
- refactor of CRUD users based on their various roles
- update documentation
- write tests
- add new features

**How should this be manually tested?**
- clone this repo, run `npm install`
- checkout to this branch `git checkout rf-user`
- run the development server, `npm run dev`
- test on Postman

**Any background context you want to provide?**
None
Any pertinent information that should be considered
None
**What are the relevant PT Story ID?**

[#173226297]https://www.pivotaltracker.com/story/show/173226297)

**Screenshots (if appropriate)**

You can have screenshots where applicable

**Questions:**

If something is unclear or you want some questions to be addressed by your peers, mention them here

### CODE REVIEW

File: constants/routesGroup.js [hunk 1]
```diff
@@ -4,15 +4,17 @@ module.exports = {
...
     ],
     // the secure routes will be much depending on app scope
     // only logged in users can access this endpoints/routes
     secureRoutes: [
         '/api/v1/auth/change-password',
-        '/api/v1/users',
-        '/api/v1/users/:userId',
-        '/api/v1/users/search/:name',
-    ]
+        // '/api/v1/users',
```
Comment (Nicanor008) [inline constants/routesGroup.js:16]: Remove unused code

**Comment by Nicanor008** [comment]
Find some time to fix this up, so as to be merged too

**Comment by Nicanor008** [comment]
Don't forget to squash your commits to only one

**Comment by neymarjimoh** [comment]
Noted

**Comment by Nicanor008** [comment]
good job on fixing the issues.
You still working on it, or everything is perfect now?

**Comment by neymarjimoh** [comment]
Everything is fine sir..

**Review by Nicanor008 — APPROVED** [review]

**Pull request merged**